src/Microsoft.ML.Core/CommandLine/CmdParser.cs

```diff
@@ -2015,6 +2016,10 @@ private void AppendSettingCore(IHostEnvironment env, StringBuilder sb, object va
                     var buffer = new StringBuilder();
                     sb.Append(GetString(env, value, buffer));
                 }
+                else if (value is InternalDataKind)
```

Comment:
This resolves some test failures due to an inconsistency in Enum serialization between .NET 6 and .NET 8.